source/contracts/reporting/ReportingWindow.sol

```diff
@@ -146,30 +139,17 @@ contract ReportingWindow is DelegationTarget, Typed, Initializable, IReportingWi
         require(markets.contains(_market));
         require(_market.getReportingTokenOrZeroByPayoutDistributionHash(_payoutDistributionHash) == msg.sender);
         IMarket.ReportingState _state = _market.getReportingState();
-        require(_state == IMarket.ReportingState.ALL_REPORTING
-            || _state == IMarket.ReportingState.LIMITED_REPORTING
+        require(_state == IMarket.ReportingState.LAST_REPORTING
+            || _state == IMarket.ReportingState.FIRST_REPORTING
             || _state == IMarket.ReportingState.DESIGNATED_REPORTING);
-        if (_state == IMarket.ReportingState.ALL_REPORTING) {
-            // always give credit for events in all-reporters phase
-            privateNoteReport(_market, _reporter);
-        } else if (_state == IMarket.ReportingState.DESIGNATED_REPORTING) {
-            privateNoteReport(_market, _reporter);
-        } else if (numberOfReportsByMarket[_market] < getMaxReportsPerLimitedReporterMarket()) {
-            // only give credit for limited reporter markets up to the max reporters for that market
-            privateNoteReport(_market, _reporter);
-        }
-        // no credit in all other cases (but user can still report)
+        reportingGasPrice.record(tx.gasprice);
```

Comment:
We should probably rename this method to `noteReportingGasPrice`.  I think we can also get rid of most of the require statements, all we really care about is just asserting that the caller actually staked some REP as part of this transaction, meaning we just need to do a provenance check on `msg.sender` to make sure it was from a reporting token (`markets.contains(_market)` and `market.isContainerForReportingToken(msg.sender)`).  I don't think we need the state checks anymore.